src/WidgetBase.ts

```diff
@@ -291,6 +293,33 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		return this._properties;
 	}
 
+	@diffProperty('registry', reference)
+	protected diffPropertyRegistry(previousProperties: any, newProperties: any): void {
+		if (this._registries.defaultRegistry === this._defaultRegistry) {
+			this._registries.remove(this._defaultRegistry);
+		}
+		const result = this._registries.replace(previousProperties.registry, newProperties.registry);
```

Comment:
Small nit, but you repeatedly access `this._registries`.  `const { _registries } = this;` would allow a build optimizations to occur and would reduce the code size.